saleor/order/utils.py

```diff
@@ -346,6 +352,18 @@ def add_variant_to_order(order, variant, quantity, discounts=None):
             ]
         )
 
+    if line.order.is_unconfirmed():
```

Comment:
What do you think to pass the flag explicitly as a function arg?
```
def add_variant_to_order(order, variant, quantity, discounts=None, allocate_stock=False):
```
IMO it should be more flexible than the current flow.

Reply:
Yep, sounds good 👍🏻